### ISSUE TITLE
would you support router macros like rocket？

### ISSUE THREAD

**Issue by Silentdoer** [http-rs/tide#627]
like title

**Comment by yoshuawuyts** [comment]
Hi @Silentdoer, thanks for opening this issue! We currently consider routing macros to be an anti-goal, as it's a different direction than where we're trying to take Tide. But I'd still be keen to learn more why you'd like these macros. Is Tide not doing something you wish it would?

**Comment by Silentdoer** [comment]
I'm sorry for my bad english, just see the ntex codes and comments:

`use ntex::web::{self, HttpServer, App};

#[ntex::main]
async fn main() -> std::io::Result<()> {
    let endpoint = "127.0.0.1:8080";
    println!("Starting server at: {:?}", endpoint);
    HttpServer::new(|| {
        // this usage is more simply
        App::new().service(test)
            .route("/test2", web::get().to(test2))
    }).bind(endpoint)?
        .run()
        .await
}

// this declare in other mod, we can clearly know the affect of test(..)
// if not have routing macros, i can't know test is get/post..., and we can set more info in this macro
#[web::get("/test/{name}")]
async fn test(name: web::types::Path<String>) -> String {
    format!("test ok {}", name)
}

async fn test2() -> &'static str {
    "test2 ok"
}`

**Comment by yoshuawuyts** [comment]
@Silentdoer Thanks for clarifying! We used to have a similar mechanism back in 2018/early 2019 but we moved away from it to what we have now. The benefits of what we have now is that routes are defined in a single file, and each endpoint has the same function signature.

There are obvious benefits to the proc-macro approach as well, but we didn't think the benefits outweighed the downsides. Hope that clarifies it somewhat!
